akka-docs/src/main/paradox/stream/operators/Source/completionStageSource.md

```diff
@@ -4,13 +4,33 @@ Streams the elements of an asynchronous source once its given *completion* opera
 
 @ref[Source operators](../index.md#source-operators)
 
+@@@ div { .group-java }
+
 ## Signature
 
+@apidoc[Source.completionStageSource](Source$) { java="#completionStageSource(java.util.concurrent.CompletionStage)" }
+
+@@@
+
 ## Description
 
 Streams the elements of an asynchronous source once its given *completion* operator completes.
 If the *completion* fails the stream is failed with that exception.
 
+@@@ div { .group-scala }
+
+For the corresponding operator for the Scala standard library `Future` see @ref:[futureSource](futureSource.md).
+
+@@@
+
+## Example
+
+Suppose we are accessing a remote service that streams user data over HTTP/2 or a WebSocket. We can model that 
+as a @apidoc[Source<User,NotUsed>](Source$) but that source will only be available once the connection has been established.
```

Comment:
```suggestion
as a @apidoc[Source[User,NotUsed]] but that source will only be available once the connection has been established.
```
^ this should look good (`<` vs `[`) and link to the right page for both Scala and Java